packages/extension/src/background/RequestBytesSign.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019 @polkadot/extension authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { KeyringPair } from '@polkadot/keyring/types';
+import { RequestSign } from './types';
+import { SignerPayloadJSON, SignerPayloadRaw } from '@polkadot/types/types';
+import { u8aToHex, hexToU8a } from '@polkadot/util';
+
+export default class RequestBytesSign implements RequestSign {
+  inner: SignerPayloadJSON | SignerPayloadRaw;
+
+  constructor (inner: SignerPayloadRaw) {
+    this.inner = inner;
+  }
+
+  sign (pair: KeyringPair): { signature: string } {
+    const inner = this.inner as SignerPayloadRaw;
+    const signedBytes = pair.sign(hexToU8a(inner.data));
```

Comment:
According to the `SignerPayloadRawBase` signature, 

```typescript
export interface SignerPayloadRawBase {
    /**
     * @description The hex-encoded data for this request
     */
    data: string;
    /**
     * @description The type of the contained data
     */
    type?: 'bytes' | 'payload';
}
```

`data` will always be hex, so it is up to the caller to pass valid data.

Also, `type?: 'bytes' | 'payload';` doesn't make sense. One can simply `const bytes = extrinsic.toU8a(true)` and `signRaw(bytes)`. Can we remove it?

Reply:
The `payload` was added specifically for other extensions, such as the centrality one - where they don't care about de-serliazing the payload, but rather just want to sign it as a raw blob. From an implementation perspective, both can just be treated as `bytes` here, it does not need to be treated separately. 

(The reason behind it - in case of transactions, the API will first try to use the normal payload sign, if that is not available will fallback to `signRaw`. Since the normal sign path is catered for, we should not receive a payload. 

Either way, it it ever does receive a `type: payload` it actually cannot quite decode it, since it is in a signable form, not a decodable form, so it should treat it as a "piece of data". (There is a small tick here which came in with v4 where the signature needs to have type information attached in the case of a `payload`, but we don't need to cater for it here)

TL;DR Ignore the type specifier on `signRaw`, we can just treat is all as bytes.

Reply:
Very interesting, thank you for writing this in an explanatory way